src/I18nProvider.js

```diff
@@ -1,39 +1,31 @@
 // Core
-import React, {useEffect} from 'react';
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
-// Utils
-import memoizeOne from 'memoize-one';
 // Context
 import I18nContext from './I18nContext';
 // Constants
-import {I18N_EVENTS} from './constants';
-// Util Hooks
-import {useForceUpdate} from './utilHooks';
+import { I18N_EVENTS } from './constants';
 
 
-const I18nProvider = ({i18n, events, children}) => {
+const I18nProvider = ({i18n, children}) => {
+  const [value, setI18n] = useState(i18n);
```

Comment:
I think this would be better using useReducer (but this is a bad appreciation because I know how it works inside the API) so keep it with useState